### PR TITLE
Move `error` crate to `packages/libs/error-stack`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

As we want to release `error-stack` this moves them outside of the hEngine package

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201893074552404/1201481007343178/f) _(internal)_

## 🔍 What does this change?

- Move `error` to `packages/libs/error-stack`
- Rename crate `error` to `error-stack`
- Adjust tests

## 📜 Does this require a change to the docs?

We probably want the top-level README.md to be adjusted cc @nonparibus 

## 🐾 Next steps

- Adjust CI to enable a stable toolchain for `error-stack` (nightly is already tested)

### CODE REVIEW

File: packages/engine/rustfmt.toml [hunk 1]
```diff
@@ -12,7 +12,6 @@ use_try_shorthand = true # Default: false
 wrap_comments = true # Default: false
 
 # Parameters
-single_line_if_else_max_width = 50 # Default: 50
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/engine/rustfmt.toml:15]: We had this set to something else, but not we use the default again. No point in specifying the default.

File: packages/libs/error-stack/Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "error"
+name = "error-stack"
 version = "0.0.0"
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/libs/error-stack/Cargo.toml:3]: I'll bump the version just before releasing

File: packages/libs/error-stack/rustfmt.toml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# General
+edition = "2021" # Default: "2015"
+unstable_features = true # Default: false
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/libs/error-stack/rustfmt.toml:3]: Pretty much everything is unstable:
- condense_wildcard_suffixes
- overflow_delimited_expr
- reorder_impl_items
- wrap_comments

- comment_width
- hex_literal_case

- format_code_in_doc_comments
- format_generated_files
- format_macro_matchers
- format_macro_bodies
- format_strings

- imports_granularity
- group_imports

[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/libs/error-stack/rustfmt.toml:3]: Sounds good then 😁

**Review by Alfred-Mountfield — APPROVED** [review]
🎉 One step closer

**Pull request merged**